### PR TITLE
feat(merkle-distributor): Make claim only callable by recipient or whitelisted claimer

### PR DESCRIPTION
**Pull request by nicholaspai**
Users shouldn't be able to claim on behalf of other users, but the [AcceleratingDistributor](https://github.com/across-protocol/across-token/pull/32) should be able to in order to make `claimAndStake` work

### CODE REVIEW

**Comment by nicholaspai** [comment]
I just realized this change would make [`claimAndStake` impossible](https://github.com/across-protocol/across-token/pull/32)

File: contracts/merkle-distributor/MerkleDistributor.sol [hunk 1]
```diff
@@ -51,10 +51,19 @@ contract MerkleDistributor is Ownable {
...
 
+    // Addresses that can claim on user's behalf. Useful to get around the requirement that claim receipient
+    // must also be claimer.
+    mapping(address => bool) public whitelistedClaimers;
+
     // Track which accounts have claimed for each window index.
     // Note: uses a packed array of bools for gas optimization on tracking certain claims. Copied from Uniswap's contract.
     mapping(uint256 => mapping(uint256 => uint256)) private claimedBitMap;
 
+    modifier onlyWhitelistedClaimer() {
```
Comment (chrismaree) [inline contracts/merkle-distributor/MerkleDistributor.sol:62]: You don't use this modifier.

File: contracts/merkle-distributor/AcrossMerkleDistributor.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+     * @param claims array of claims to claim.
+     */
+    function claimMulti(Claim[] memory claims) public override {
+        if (!whitelistedClaimers[msg.sender]) {
+            uint256 claimCount = claims.length;
+            for (uint256 i = 0; i < claimCount; i++) {
+                require(claims[i].account == msg.sender, "invalid claimer");
+            }
+        }
+        super.claimMulti(claims);
```
Comment (chrismaree) [inline contracts/merkle-distributor/AcrossMerkleDistributor.sol:53]: assuming `claimMulti` and `claim` from within the MerkleDistributor works correctly, this wrapper contract looks perfect to me!

File: contracts/merkle-distributor/AcrossMerkleDistributor.sol [hunk 3]
```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@uma/core/contracts/merkle-distributor/implementation/MerkleDistributor.sol";
+
+/**
+ *
```
Comment (chrismaree) [inline contracts/merkle-distributor/AcrossMerkleDistributor.sol:7]:
```suggestion
```

**Review by chrismaree — APPROVED** [review]
LGTM!

File: contracts/merkle-distributor/AcrossMerkleDistributor.sol (hunk 2 above)
Comment (nicholaspai) [reply contracts/merkle-distributor/AcrossMerkleDistributor.sol:53]: yup and the tests give me confidence, which i didn't modify and should work against the parent's behavior

**Pull request merged**